// navigation/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var reactRouterDom = require('react-router-dom');

var style = {"nav":"_2Aors","pointer":"_zD9lR","form":"_It3qu"};

function CarteMenuNav(_ref) {
  var groupe = _ref.groupe,
      index = _ref.index,
      setSelect = _ref.setSelect;
  return /*#__PURE__*/React__default.createElement("article", {
    className: style.nav
  }, /*#__PURE__*/React__default.createElement("div", {
    className: style.form
  }, /*#__PURE__*/React__default.createElement("div", {
    onClick: function onClick() {
      setSelect(index);
    },
    className: style.pointer
  }, groupe)));
}

function CarteNav(_ref) {
  var navigation = _ref.navigation;
  var area = navigation[1] !== null ? "/" + navigation[1] : "";
  var nom = "/" + navigation[0];
  var lien = area + nom;
  return /*#__PURE__*/React__default.createElement("article", {
    className: style.nav
  }, /*#__PURE__*/React__default.createElement("div", {
    className: style.form
  }, /*#__PURE__*/React__default.createElement(reactRouterDom.Link, {
    to: lien,
    className: style.pointer
  }, navigation[0])));
}

var style$1 = {"displayNav":"_x0hTR"};

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];

  return arr2;
}

function _createForOfIteratorHelperLoose(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it) return (it = it.call(o)).next.bind(it);

  if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it) o = it;
    var i = 0;
    return function () {
      if (i >= o.length) return {
        done: true
      };
      return {
        done: false,
        value: o[i++]
      };
    };
  }

  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

var getNavigation = function getNavigation(data) {
  var navTab = {};

  for (var _iterator = _createForOfIteratorHelperLoose(data), _step; !(_step = _iterator()).done;) {
    var d = _step.value;

    if (!navTab[d.groupe]) {
      navTab[d.groupe] = [];
    }

    navTab[d.groupe] = [].concat(navTab[d.groupe], [[d.nom, d.adresse]]);
  }

  return navTab;
};

var navigation = {
  groupe: [],
  load: function load(fetchRequest, contentObject, loader) {
    fetchRequest(JSON.stringify(contentObject), function (data) {
      console.log(data);
      navigation.groupe = getNavigation(data);
      loader(true);
    });
  }
};

function NavigationMenu(_ref) {
  var fetchRequest = _ref.fetchRequest,
      contentObject = _ref.contentObject;

  var _useState = React.useState(-1),
      select = _useState[0],
      setSelect = _useState[1];

  var _useState2 = React.useState(false),
      load = _useState2[0],
      loader = _useState2[1];

  React.useEffect(function () {
    navigation.load(fetchRequest, contentObject, loader);
    alert('nav');
    console.log(navigation);
  }, []);
  var navListe = navigation.groupe;
  return /*#__PURE__*/React__default.createElement("div", {
    className: style$1.displayNav
  }, load ? Object.entries(navListe).map(function (groupe, index) {
    if (groupe[1].length === 1) {
      return /*#__PURE__*/React__default.createElement(CarteNav, {
        navigation: groupe[1][0],
        key: index
      });
    } else {
      return /*#__PURE__*/React__default.createElement("div", {
        key: index,
        className: style$1.displayNav
      }, select !== index ? /*#__PURE__*/React__default.createElement(CarteMenuNav, {
        setSelect: setSelect,
        index: index,
        groupe: groupe[0]
      }) : groupe[1].map(function (nav, indexM) {
        return /*#__PURE__*/React__default.createElement(CarteNav, {
          key: indexM,
          navigation: nav
        });
      }));
    }
  }) : null);
}

var NavigationDisplay = function NavigationDisplay(_ref) {
  var fetchRequest = _ref.fetchRequest,
      contentObject = _ref.contentObject;
  return /*#__PURE__*/React__default.createElement(NavigationMenu, {
    fetchRequest: fetchRequest,
    contentObject: contentObject
  });
};
var NavigationBrowser = function NavigationBrowser(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/React__default.createElement(reactRouterDom.BrowserRouter, {
    children: children
  });
};
var NavigationRouter = function NavigationRouter(_ref3) {
  var children = _ref3.children,
      link = _ref3.link;
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    path: link,
    children: children
  });
};
var NavigationRouterExact = function NavigationRouterExact(_ref4) {
  var children = _ref4.children,
      link = _ref4.link;
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Route, {
    exact: true,
    path: link,
    children: children
  });
};
var NavigationRedirect = function NavigationRedirect(_ref5) {
  var link = _ref5.link;
  return /*#__PURE__*/React__default.createElement(reactRouterDom.Redirect, {
    to: link
  });
};

exports.NavigationBrowser = NavigationBrowser;
exports.NavigationDisplay = NavigationDisplay;
exports.NavigationRedirect = NavigationRedirect;
exports.NavigationRouter = NavigationRouter;
exports.NavigationRouterExact = NavigationRouterExact;
