// button/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));

var style = {"button":"_1tCdz","Create":"_1KH-U","Delete":"_2oNlu","Back":"_3ocOo","Edit":"_1z-tM"};

function ButtonSite(_ref) {
  var type = _ref.type,
      text = _ref.text,
      oc = _ref.oc;
  return /*#__PURE__*/React.createElement("button", {
    className: style.button + " " + style[type],
    onClick: function onClick() {
      return oc();
    }
  }, text);
}

var ButtonCustom = function ButtonCustom(_ref) {
  var text = _ref.text,
      type = _ref.type,
      callback = _ref.callback;
  return /*#__PURE__*/React.createElement(ButtonSite, {
    type: type,
    text: text,
    oc: callback
  });
};
var create = 'Create';
var edit = 'Edit';
var deleteItem = 'Delete';
var back = 'Back';
var ButtonTypeList = {
  create: create,
  edit: edit,
  deleteItem: deleteItem,
  back: back
};

exports.ButtonCustom = ButtonCustom;
exports.ButtonTypeList = ButtonTypeList;
