// toaster/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

var style = {"toaster":"_382jh","toasterHigh":"_33VHD","toasterBottom":"_3Swwu","error":"_2Ecqe","success":"_2tmAG","info":"_1J-yJ","right":"_INbaN","left":"_-Sw8h"};

function Toaster(_ref) {
  var title = _ref.title,
      text = _ref.text,
      position = _ref.position,
      color = _ref.color;
  var toasterClass = style.toaster + ' ' + style[position];
  var highClass = style.toasterHigh + ' ' + style[color];
  var bottomClass = style.toasterBottom;
  return /*#__PURE__*/React__default.createElement("div", {
    className: toasterClass
  }, /*#__PURE__*/React__default.createElement("div", {
    className: highClass
  }, title), /*#__PURE__*/React__default.createElement("div", {
    className: bottomClass
  }, text));
}

var toastersList = new Array();
var numberofToasters = 0;

var setter = function setter() {};

var NewToaster = function NewToaster(_ref) {
  var title = _ref.title,
      text = _ref.text,
      position = _ref.position,
      color = _ref.color;
  toastersList.push({
    toaster: function toaster() {
      return /*#__PURE__*/React__default.createElement(Toaster, {
        title: title,
        text: text,
        position: position,
        color: color
      });
    },
    index: numberofToasters
  });
  numberofToasters++;
  setter();
};
var right = 'right';
var left = 'left';
var positionToaster = {
  right: right,
  left: left
};
var error = 'error';
var success = 'success';
var info = 'info';
var colorToaster = {
  error: error,
  success: success,
  info: info
};

var ToastersDisplay = function ToastersDisplay() {
  var _useState = React.useState(false),
      nb = _useState[0],
      del = _useState[1];

  setter = function setter() {
    return del(!nb);
  };

  return /*#__PURE__*/React__default.createElement("div", null, toastersList.map(function (Toaster, index) {
    return /*#__PURE__*/React__default.createElement("div", {
      key: index,
      onClick: function onClick() {
        toastersList = toastersList.filter(function (x) {
          return x.index !== Toaster.index;
        });
        del(!nb);
      }
    }, /*#__PURE__*/React__default.createElement(Toaster.toaster, null));
  }));
};

exports.NewToaster = NewToaster;
exports.colorToaster = colorToaster;
exports.default = ToastersDisplay;
exports.positionToaster = positionToaster;
