// card/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));

var style = {"article":"_KKfTP","pointer":"_2_jTR","form":"_2Mp25","containerBorder":"_1iQla","projets":"_3GuRH"};

function CardForm(_ref) {
  var card = _ref.card,
      children = _ref.children;
  return /*#__PURE__*/React.createElement("article", {
    className: style.article
  }, /*#__PURE__*/React.createElement("form", {
    action: card.action,
    method: card.method,
    onSubmit: function onSubmit(e) {
      return card.submit(e);
    },
    className: style.form
  }, /*#__PURE__*/React.createElement("button", {
    className: style.pointer
  }, /*#__PURE__*/React.createElement("div", {
    className: style.containerBorder,
    children: children
  }))));
}

function CardSimple(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React.createElement("article", {
    className: style.article + ' ' + style.projets
  }, /*#__PURE__*/React.createElement("div", {
    className: style.form
  }, /*#__PURE__*/React.createElement("div", {
    className: style.pointer
  }, /*#__PURE__*/React.createElement("div", {
    className: style.containerBorder,
    children: children
  }))));
}

var FormCard = function FormCard(_ref) {
  var card = _ref.card,
      children = _ref.children;
  return /*#__PURE__*/React.createElement(CardForm, {
    card: card,
    children: children
  });
};
var cardByDefault = {
  action: '/',
  method: 'GET',
  submit: function submit() {
    return console.log('Data Submit');
  }
};
var SimpleCard = function SimpleCard(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/React.createElement(CardSimple, {
    children: children
  });
};

exports.FormCard = FormCard;
exports.SimpleCard = SimpleCard;
exports.cardByDefault = cardByDefault;
