// header/dist/index.modern.js
import React from 'react';

var style = {"contact":"_3fkGl","titre":"_ziF6X","header":"_YWPXt","ConnexionDiv":"_Qj5X4","h100":"_1-nxu","ConnectedDiv":"_dLzDs","containerTitre":"_w82ez","deconnexion":"_3zmrB"};

function Header(_ref) {
  var contact = _ref.contact,
      title = _ref.title,
      connect = _ref.connect,
      resource = _ref.resource;
  return /*#__PURE__*/React.createElement("header", {
    id: style.header
  }, /*#__PURE__*/React.createElement("div", {
    id: style.contact
  }, contact.map(function (element, index) {
    if (element.type === 'Link') {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, /*#__PURE__*/React.createElement("a", {
        href: element.link
      }, element.text));
    } else {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, /*#__PURE__*/React.createElement("span", {
        key: index
      }, element.text));
    }
  })), /*#__PURE__*/React.createElement("div", {
    id: style.containerTitre
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", {
    id: style.titre
  }, title.title), /*#__PURE__*/React.createElement("h3", null, title.subtitle)), connect.connected ? connect.connectedComponentbyDefault ? /*#__PURE__*/React.createElement("div", {
    className: style.h100
  }, /*#__PURE__*/React.createElement("div", {
    id: style.ConnectedDiv
  }, /*#__PURE__*/React.createElement("span", null, connect.pseudo[0])), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("a", {
    href: connect.deconnexion,
    className: style.deconnexion
  }, "D\xE9connexion"))) : /*#__PURE__*/React.createElement("div", null, connect.connectedComponent()) : connect.connexionComponentbyDefault ? /*#__PURE__*/React.createElement("div", {
    id: style.ConnexionDiv
  }, /*#__PURE__*/React.createElement("form", {
    action: connect.connexionURL,
    method: "get"
  }, /*#__PURE__*/React.createElement("button", null, /*#__PURE__*/React.createElement("span", null, resource.list.connected), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("span", null, resource.list.slash), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("span", null, resource.list.subscribe)))) : /*#__PURE__*/React.createElement("div", null, connect.connexionComponent())));
}

var HeaderComponent = function HeaderComponent(_ref) {
  var contact = _ref.contact,
      title = _ref.title,
      connect = _ref.connect,
      resource = _ref.resource;
  return /*#__PURE__*/React.createElement(Header, {
    contact: contact,
    title: title,
    connect: connect,
    resource: resource
  });
};
var contactByDefault = [{
  type: 'Text',
  text: 'Contact'
}, {
  type: 'Link',
  text: 'Link',
  link: '#Link'
}];
var titleByDefault = {
  title: 'Title 1',
  subtitle: 'SubTitle 1'
};
var connectByDefault = {
  connected: true,
  connexionComponentbyDefault: true,
  connexionURL: '/Connexion/',
  connexionComponent: function connexionComponent() {},
  connectedComponentbyDefault: true,
  pseudo: 'Granguil',
  deconnexion: '/Connexion/Deconnexion',
  connectedComponent: function connectedComponent() {}
};

export default HeaderComponent;
export { connectByDefault, contactByDefault, titleByDefault };
